docs/4.3/aws_oss_guide.md

```diff
@@ -386,12 +379,308 @@ To upgrade to a newer version of Teleport:
+# (Optional) Verify the downloaded binary with the SHA-256 sum for your binary.
+curl -o kubectl.sha256 https://amazon-eks.s3.us-west-2.amazonaws.com/1.16.8/2020-04-16/bin/linux/amd64/kubectl.sha256
+# Check the SHA-256 sum for your downloaded binary.
+openssl sha1 -sha256 kubectl
+# Apply execute permissions to the binary.
+chmod +x ./kubectl
+# Copy the binary to a folder in your PATH. If you have already installed a version of kubectl, then we recommend creating a $HOME/bin/kubectl and ensuring that $HOME/bin comes first in your $PATH.
+sudo mv ./kubectl /usr/local/bin
+# After you install kubectl, you can verify its version with the following command:
+kubectl version --short --client
+```
+
+Create kubeconfig:
+```bash
+aws eks update-kubeconfig --name [teleport-cluster] --region us-west-2
+# Added new context arn:aws:eks:us-west-2:480176057099:cluster/teleport-eks-cluster to /home/ec2-user/.kube/config
+```
+
+Create SSL certificate for HTTPs. It is absolutely crucial to properly configure TLS
+for HTTPS when you use Teleport Proxy in production. For simplicity, we are using Let’s
+Encrypt to issue certificates and simple DNS resolution.
+
+However, using an Elastic IP and a Route53 domain name would be appropriate for production use cases.
+
+Install certbot:
+
+```bash
+# Install the EPEL release package for RHEL 7 and enable the EPEL repository.
+sudo yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+sudo yum install -y certbot python-certbot-nginx
+```
+
+Run Certbot
+```bash
+export TELEPORT_PUBLIC_DNS_NAME=[teleport-proxy-url]
+echo $TELEPORT_PUBLIC_DNS_NAME
+export EMAIL=[email-for-letsencrypt]
+
+sudo certbot certonly --standalone \
+             --preferred-challenges http \
+                      -d $TELEPORT_PUBLIC_DNS_NAME \
+                      -n \
+                      --agree-tos \
+                      --email=$EMAIL
+```
+
+#### 4. Setup Github Auth
+
+Run this on the Teleport EC2 Host, see [Github Auth](admin-guide.md#github-oauth-20) for more info.
+
+```bash
+export TELEPORT_PUBLIC_DNS_NAME="[teleport-proxy-url]"
+export GH_CLIENT_ID="[github-client-id]"
+export GH_SECRET="[github-oauth-secret]"
+export GH_ORG="[github-org]"
+export GH_TEAM="[github-team]"
+
+cat > github.yaml << EOF
+kind: github
+version: v3
+metadata:
+  # connector name that will be used with 'tsh --auth=github login'
+  name: github
+spec:
+  # client ID of Github OAuth app
+  client_id: $GH_CLIENT_ID
+  # client secret of Github OAuth app
+  client_secret: $GH_SECRET
+  # connector display name that will be shown on web UI login screen
+  display: Github
+  # callback URL that will be called after successful authentication
+  redirect_url: https://$TELEPORT_PUBLIC_DNS_NAME:3080/v1/webapi/github/callback
+  # mapping of org/team memberships onto allowed logins and roles
+  teams_to_logins:
+  - kubernetes_groups:
+    - system:masters
```

Comment:
Add a comment here: `# change this to restrict the access of users logging in via GitHub; system:masters gives everyone full access to the entire cluster`